Fails pep8 test

kmip/services/kmip_client.py:228:15: F821 undefined name 'e'

```
        for host in self.host_list:
            self.host = host
            sock = socket.socket(socket.AF_INET, socket.SOCK_STREAM)
            self._create_socket(sock)
            self.protocol = KMIPProtocol(self.socket)
            try:
                self.socket.connect((self.host, self.port))
            except Exception as e:
                self.logger.error("An error occurred while connecting to "
                                  "appliance " + self.host)
                self.socket.close()
            else:
                return

        self.socket = None
        raise e
```
Obviously on the last line "raise e", e is not in scope. I assume the last two lines should be moved into the expect block.

Yes I was running "tox -e pep8" and got the following error:
kmip/services/kmip_client.py:228:15: F821 undefined name 'e'

I believe the problem isn't the "for" loop. It is that "e" isn't assigned until it enters the exception handler (except Exception as e).

--- a/kmip/services/kmip_client.py
+++ b/kmip/services/kmip_client.py
@@ -210,6 +210,8 @@ class KMIPProxy(KMIP):
         self.logger.debug("KMIPProxy suppress_ragged_eofs: {0}".format(
             self.suppress_ragged_eofs))
 
+        last_error = None
+
         for host in self.host_list:
             self.host = host
             sock = socket.socket(socket.AF_INET, socket.SOCK_STREAM)
@@ -221,11 +223,13 @@ class KMIPProxy(KMIP):
                 self.logger.error("An error occurred while connecting to "
                                   "appliance " + self.host)
                 self.socket.close()
+                last_error = e
             else:
                 return
 
         self.socket = None
-        raise e
+        if last_error:
+            raise last_error
 
     def _create_socket(self, sock):
         self.socket = ssl.wrap_socket(

